fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java

```diff
@@ -0,0 +1,185 @@
+        if (jsonAvroSchema == null) {
+            return false;
+        }
+
+        // This may throw a NullPointerException, if the AVRO schema file does not have a `fields` JSON object.
+        int fieldSize = jsonAvroSchema.get("fields").size();
+
+        jsonAvroSchema = jsonAvroSchema.get("fields");
+
+        HashSet<String> avroFields = new HashSet<>(fieldSize);
+
+        // Streaming cannot be used, because `avroFields` is a JsonNode object
+        for (int i = 0 ; i < fieldSize; i++) {
+            // Parsing the JsonNode object as Text, `toString()` adds an additional `"`
+            avroFields.add(jsonAvroSchema.get(i).get("name").asText());
+        }
+
+        Function<DimensionField, String> curriedConverter = curry(dimension, dimensionFieldMapper);
+
+        // Generates a map of configured dimension field names and the converted field names, for easier lookups in the
+        // `parseAvroFileDimensionRows()`
+        convertedDimensionFields = dimension.getDimensionFields()
+                .stream()
+                .collect(Collectors.toMap(dimensionField -> dimensionField.getName(), curriedConverter));
+
+        // Returns true, only if all the dimension fields present in the configured dimension exist in the AVRO schema
+        return convertedDimensionFields.values().containsAll(avroFields);
+    }
+
+    /**
+     * Parses the avro file and populates the dimension rows after validating the schema.
+     *
+     * @param dimension The dimension object used to configure the dimension
+     * @param avroFilePath The path of the AVRO data file (.avro)
+     * @param avroSchemaPath The path of the AVRO schema file (.avsc)
+     *
+     * @throws NullPointerException thrown if JSON object `fields` is not present
+     * @throws IOException thrown if there is error parsing the avro files
+     */
+    public void parseAvroFileDimensionRows(Dimension dimension, String avroFilePath, String avroSchemaPath)
+        throws NullPointerException, IOException {
+
+        if (isSchemaValid(dimension, avroSchemaPath)) {
+
+            JsonNode jsonAvroSchema = convertFileToJsonNode(avroSchemaPath);
+
+            // Creates an AVRO schema object based on the parsed AVRO schema file (.avsc)
+            Schema schema = new Schema.Parser().parse(jsonAvroSchema.toString());
+
+            // Creates an AVRO DatumReader object based on the AVRO schema object
+            DatumReader<GenericRecord> datumReader = new GenericDatumReader<>(schema);
+
+            // Creates an AVRO DataFileReader object that reads the AVRO data file one record at a time
+            DataFileReader<GenericRecord> dataFileReader =
+                new DataFileReader<>(new File(avroFilePath), datumReader);
+
+            // Represents each record in the avro data file
+            GenericRecord avroData = null;
+
+            while (dataFileReader.hasNext()) {
+                avroData = dataFileReader.next(avroData);
```

Comment:
Is there a strong reason to use this mechanism instead of a more basic foreach loop? (eg. `for (GenericRecord genericRecord : dataFileReader) {`)